reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java

```diff
@@ -87,16 +91,40 @@ public String getFiscalEventDetailsInformation(List<FiscalEvent> fiscalEventList
     }
 
     /**
-     * @return
+     * It calculates time from cron job expression <strong>samplejob.frequency</strong>.
+     * It also check expression length - which is strict 6, any other length value will end up in exception.
+     *
+     * @return it returns in millisecond with additional overlap minutes.
      */
-    public Long getValidSearchIntervalTime() {
-        String plainTime = applicationConfiguration.getIfixFiscalEventSearchTimeIntervalMinutes();
+    public @NotNull Long translateCronExpressionIntoMilliSecond() {
+        String plainTime = applicationConfiguration.getIfixFiscalEventSearchTimeOverlapMinutes();
 
         if (!org.apache.commons.lang3.StringUtils.isNumeric(plainTime)) {
-            log.error(">>>>> Ifix fiscal event search interval time value is invalid");
-            throw new GenericCustomException(TIME, "Ifix fiscal event search interval time value is invalid");
+            log.error(">>>>> Ifix fiscal event search overlap time value is invalid");
+            throw new GenericCustomException(TIME, "Ifix fiscal event search overlap time value is invalid");
         }else {
-            return Long.parseLong(plainTime) * 60 * 1000;
+            Long overlapTime = Long.parseLong(plainTime) * 60 * 1000;
+
+            String cronExpression = applicationConfiguration.getSampleJobFrequency();
+            String[] fields = StringUtils.tokenizeToStringArray(cronExpression, " ");
+
+            if (fields == null || fields.length != 6) {
+                log.error(String.format(">>>>> Cron expression must consist of 6 fields (found %d in \"%s\")",
+                        fields.length, cronExpression));
+
+                throw new GenericCustomException("CRON_EXPRESSION",
```

Comment:
Instead of throwing an error, can we just drop the last part and move ahead. Just add a log statement that considering only the first 6 parts and dropping the 7th element.

Reply:
I have written for 6 and more than six and working fine but while testing other scenarios come like empty, null and less than 6 - all these scenario throws exception.

`if (fields == null || fields.length < 6) {
            log.error(String.format("Cron expression length is not valid"));
            throw new GenericCustomException("CRON_EXPRESSION", "Cron expression length is not valid");
        }`
Should I put that condition and throw exception.

Reply:
There won't be null, empty string. In those scenarios, the quartz cron job will also fail. No need to handle those.

Reply:
Don't throw any error. Just log that you are dropping the 7th element.

Reply:
Done